deltachat-ffi/deltachat.h

```diff
@@ -3558,6 +3616,41 @@ char*           dc_msg_get_filename           (const dc_msg_t* msg);
 char*           dc_msg_get_filemime           (const dc_msg_t* msg);
 
 
+/**
+ * Return file from inside an webxdc message.
+ *
+ * @memberof dc_msg_t
+ * @param msg The webxdc instance.
+ * @param filename The name inside the archive,
+ *     must be given as a relative path (no leading `/`).
+ * @param ret_bytes Pointer to a size_t. The size of the blob will be written here.
+ * @return The blob must be released using dc_str_unref() after usage.
+ *     NULL if there is no such file in the archive or on errors.
+ */
+char*             dc_msg_get_webxdc_blob      (const dc_msg_t* msg, const char* filename, size_t* ret_bytes);
+
+
+/**
+ * Get info a webxdc message, in JSON format.
+ * The returned JSON string has the following key/values:
+ *
+ * - name: The name of the app.
+ *   Defaults to the filename if not set in the manifest.
+ * - icon: App icon file name.
+ *   Defaults to an standard icon if nothing is set in the manifest.
```

Comment:
do we need it in the manifest or can we just say have the icon named `icon.png` and we find it automatically without manifest?

Reply:
i was also thinking about that, however, i have the _feeling_, all in all, it is better to make it explicit. sooner or later we might need an option for $reasons - and then we have to handle both.

already now, we allow .png and .jpg - so we have to probe already against two possibilities.

Reply:
k, we shift to simple icon.png or icon.jpg at https://github.com/deltachat/deltachat-core-rust/pull/2952 , seems a bit easier for both, implementations and webxdc-devs.

Reply:
this docs need to be updated to say the icon is not get from manifest